### DISCUSSION TITLE
How do you define a job that only materializes unmaterialized assets?

### DISCUSSION THREAD

**Discussion by JoostvDoorn** [dagster-io/dagster#18154] [Q&A]
When defining a job that materializes some assets, I want to be able to run certain assets only if they haven't been materialized yet. Is there anyway that I can define this job easily? e.g.:
```python
@dagster.asset
def parent_asset():
   return 1

@dagster.asset
def some_asset(parent_asset):
   return 0

dagster.define_asset_job(
    "some_job",
    dagster.AssetSelection.keys(
        "some_asset",
    ).upstream(),
    partitions_def=experiments_partitions_def,
)
```

Here I want to run all upstream assets of `some_asset` but if `parent_asset` is already materialized how would I be able to avoid materializing this asset? Ideally some mechanism that will allow me define which assets will be skipped if they already exist, and some will not be skipped.

**Comment by benpankow** [comment]
Hi Joost - could you elaborate a bit on what your desired end result is here? Currently there's not a way to specify a job definition that does this, but you can selectively subset that job definition. You might be able to use a sensor which would run periodically and launch runs with an [`asset_selection`](https://docs.dagster.io/_apidocs/schedules-sensors#dagster.RunRequest.asset_selection) to specify only unmaterialized assets.

One other option might be our [experimental auto-mateiralization functionality](https://docs.dagster.io/concepts/assets/asset-auto-execution) which aims to solve situations like this where you want to keep a set of assets 'up-to-date' but not re-run assets that have already been materialized.
